charts_common/lib/src/chart/cartesian/axis/axis_tick.dart

```diff
@@ -100,11 +100,24 @@ class AxisTicks<D> extends Tick<D> implements Comparable<AxisTicks<D>> {
   /// Linearly interpolate between two numbers.
   ///
   /// From lerpDouble in dart:ui which is Flutter only.
-  double? _lerpDouble(double? a, double? b, double t) {
-    if (a == null && b == null) return null;
+  double? _lerpDouble(num? maybeA, num? maybeB, double t) {
+    var a = maybeA;
+    var b = maybeB;
+    if (a == b || (a?.isNaN ?? false) && (b?.isNaN ?? false)) {
+      return a?.toDouble();
+    }
     a ??= 0.0;
     b ??= 0.0;
-    return a + (b - a) * t;
```

Comment:
These are the same, btw.   Just modern flutter arrived at a different formulation of the same equation:
a + (b - a) * t
a + (bt - at)
a - at  +  bt
a(1-t) + bt